compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs

```diff
@@ -4,271 +4,102 @@
+    RenderList items -> spaced (map (bullet . renderRst env) items)
+    RenderFields fields -> renderRstFields env fields
+    RenderPara text -> [renderRstText env text]
+    RenderDocs docText -> docTextToRst docText
+    RenderAnchor anchor -> [".. _" <> unAnchor anchor <> ":"]
+
+renderRstText :: RenderEnv -> RenderText -> T.Text
+renderRstText env = \case
+    RenderConcat ts -> mconcatMap (renderRstText env) ts
+    RenderPlain text -> text
+    RenderStrong text -> T.concat ["**", text, "**"]
+    RenderLink anchor text ->
+        case lookupAnchor env anchor of
+            Nothing -> text
+            Just _ -> T.concat ["`", text, " <", unAnchor anchor, "_>`_"]
+    RenderDocsInline docText ->
+        T.unwords (docTextToRst docText)
+
+-- Utilities
+
+-- | Put an extra newline in between chunks. Because Rst has support for definition
+-- lists, we *sometimes* don't put a newline in between, in particular when the
+-- next line is indented and looks like a type signature or definition. This
+-- should affect the output very little either way (it's only spacing).
+spaced :: [[T.Text]] -> [T.Text]
+spaced = intercalate [""] . respace
   where
-    title = "Module " <> unModulename md_name
-
-    section :: T.Text -> (a -> RenderOut) -> [a] -> RenderOut
-    section _ _ [] = mempty
-    section sectionTitle f xs = mconcat
-        [ renderLines
-            [ ""
-            , sectionTitle
-            , T.replicate (T.length sectionTitle) "^"
-            ]
-        , mconcatMap f xs
-        ]
-
-renderTemplateDoc :: TemplateDoc -> RenderOut
-renderTemplateDoc TemplateDoc{..} = mconcat $
-    [ renderAnchor td_anchor
-    , renderLineDep $ \env -> T.unwords . concat $
-        [ [bold "template"]
-        , renderContext env td_super
-        , [makeAnchorLink env td_anchor (unTypename td_name)]
-        , td_args
-        ]
-    , maybe mempty ((renderLine "" <>) . renderIndent 2 . renderDocText) td_descr
-    , renderLine ""
-    , renderIndent 2 (fieldTable td_payload)
-    , renderLine ""
-    ] ++ map (renderIndent 2 . renderChoiceDoc) td_choices
-
-renderTemplateInstanceDoc :: TemplateInstanceDoc -> RenderOut
-renderTemplateInstanceDoc TemplateInstanceDoc{..} = mconcat
-    [ renderAnchor ti_anchor
-    , renderLinesDep $ \env ->
-        [ "template instance " <>
-            makeAnchorLink env ti_anchor (unTypename ti_name)
-        , "  = " <> renderType env ti_rhs
-        , ""
-        ]
-    , maybe mempty ((<> renderLine "") . renderIndent 2 . renderDocText) ti_descr
-    ]
-
-renderChoiceDoc :: ChoiceDoc -> RenderOut
-renderChoiceDoc ChoiceDoc{..} = mconcat
-    [ renderLine $ prefix "+ " $ bold $ "Choice " <> unTypename cd_name
-    , maybe mempty ((renderLine "" <>) . renderIndent 2 . renderDocText) cd_descr
-    , renderIndent 2 (fieldTable cd_fields)
+    respace = \case
+        [line1] : (line2 : block) : xs
+            | "`" `T.isPrefixOf` line1
+            || "**type**" `T.isPrefixOf` line1
```

Comment:
I just spent a minute researching since when you can nest guards as `||` in Haskell until I realized by stupidity :)
How about either don’t align `||` with the guards or use `any (`T.isPrefixOf` line1) ["**template instance**, " :", " ="]` which I personally find easier to understand.

Reply:
That's better :-) thanks